### DISCUSSION TITLE
TextRange indicates outside of text which has linebreak at last.

### DISCUSSION THREAD

**Discussion by whistyun** [AvaloniaUI/Avalonia#6454] [Q&A] [answered]
**Describe**

When text end with linebreak is inputed to `TextLayout`, the last `TextRange` indicates outside of the text. Is this as intended? 

**To Reproduce**

run the bellow code using Avalonia v0.10.6.

*code:*  
```cs
var text = "asdf\n";

var typeface = new Typeface(
        FontFamily.Default,
        FontStyle.Normal,
        FontWeight.Normal);

var layout = new TextLayout(
    text,
    typeface,
    12f,
    new SolidColorBrush(Colors.White),
    textWrapping: TextWrapping.Wrap);

foreach (var range in layout.TextLines.Select(ln => ln.TextRange))
{
    Debug.WriteLine($"Start:{range.Start}, End:{range.End}, Length:{range.Length}");
    // text.Substring(range.Start, range.Length);
}
```

*outout:*  
```txt
Start:0, End:4, Length:5
Start:5, End:5, Length:1
```

**Comment by Gillibald** [comment, marked as answer]
There is always an empty line added to the TextLayout if the text ends with a line break.

**Reply by Gillibald** [reply]
https://github.com/AvaloniaUI/Avalonia/pull/4893 uses this. Otherwise you will not by able to position the caret properly after the line break.

**Reply by Gillibald** [reply]
To sum it up everything works as intended.
